### PR TITLE
Collect command name from procfs

### PR DESCRIPTION
**Pull request by robertjli**
### What does this PR do?

Collect the command name from `/proc/[pid]/comm` ([man page](https://man7.org/linux/man-pages/man5/proc.5.html#:~:text=you%20to%20see.-,/proc/pid/comm,-(since%20Linux%202.6.33))).

### Motivation

Collecting the command name will enable more accurate language detection. For example, some Ruby-based commands do not contain any reference to Ruby, but the `comm` file contains the name `ruby`.

### Describe how to test/QA your changes

Run a manual process check using `process-agent check process`, and ensure that the `Command Name` field exists and is populated.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/process/procutil/process_linux.go [hunk 1]
```diff
@@ -364,6 +366,20 @@ func (p *probe) getCmdline(pidPath string) []string {
 	return trimAndSplitBytes(cmdline)
 }
 
+// getCommandName retrieves the command name from "comm" file for a process in procfs
+func (p *probe) getCommandName(pidPath string) string {
```
Comment (just-chillin) [inline pkg/process/procutil/process_linux.go:370]: Would you mind quickly adding a test for this? Just want to get our code coverage up 😄

File: releasenotes/notes/process-comm-e0a76997f81d0542.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    During the process check, collect command name from `/proc/[pid]/comm`.
```
Comment (xornivore) [inline releasenotes/notes/process-comm-e0a76997f81d0542.yaml:11]: Would be good to be more explicit about the intention here.

File: pkg/process/procutil/process_linux.go (hunk 1 above)
Comment (robertjli) [reply pkg/process/procutil/process_linux.go:370]: I added a basic test with a hardcoded check against one of the fake procfs entries. I think it serves its purpose, but let me know if you had something else in mind.

Note that most tests for this class's methods test the output against the [gopsutil library](https://github.com/DataDog/gopsutil), but `comm` is not collected by gopsutil.

File: releasenotes/notes/process-comm-e0a76997f81d0542.yaml (hunk 2 above)
Comment (robertjli) [reply releasenotes/notes/process-comm-e0a76997f81d0542.yaml:11]: What do you think about
> During the process check, collect command name from `/proc/[pid]/comm`. This allows more accurate language detection of processes.

Comment (xornivore) [reply releasenotes/notes/process-comm-e0a76997f81d0542.yaml:11]: This is great, thank you 🙇

File: releasenotes/notes/process-comm-e0a76997f81d0542.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    During the process check, collect command name from `/proc/[pid]/comm`. This
```
Comment (urseberry) [inline releasenotes/notes/process-comm-e0a76997f81d0542.yaml:11]:
```suggestion
    During the process check, collect the command name from `/proc/[pid]/comm`. This
```

**Review by urseberry — APPROVED** [review]

File: releasenotes/notes/process-comm-e0a76997f81d0542.yaml (hunk 3 above)
Comment (robertjli) [reply releasenotes/notes/process-comm-e0a76997f81d0542.yaml]: Thank you!

**Review by just-chillin — APPROVED** [review]

**Pull request merged**